File: readability.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int words(string sentences);
int letters(string sentences);
int nsentence(string sentences);


int main(void)
{
//getting input from the user
    string sentences = get_string("enter the sentences: ");
    //printf(" letters: %i words: %i sentences: %i \n", letters(sentences), words(sentences), nsentence(sentences));

//calculate the letters per 100 words and sentences per 100 word;
    float avg_letter = ((float) letters(sentences)/(float) words(sentences))*100;
    //printf("letters_avg: %f", avg_letter);


    float avg_sntcs = ((float) nsentence(sentences)/(float) words(sentences))*100.00000;
    //printf("avg_sntcs: %f", avg_sntcs);

//putting the value of the avg_letter and avg_sntcs; in the grade formula;
    float index = 0.0588 * avg_letter - 0.296 * avg_sntcs - 15.8;
    //printf("index: %f \n", index);



    int grade =((int) round (index));
    //printing the grade level;
    if(grade <= 1 )
    {
        printf("Before Grade 1 \n");
    }
    else if(grade >= 16)
    {
        printf("Grade 16+ \n");
    }
    else if(grade > 1 && grade < 16)
    {
        printf("Grade %i \n", grade);
    }
    else
    {
        return 10;
    }

}

 //checking for how many letter the user has typed
int letters (string sentences)
{
    int  m = 0;
    for(int i = 0; i < strlen(sentences); i++)
    {
        if((sentences[i]>= 'a' && sentences[i]<= 'z') || (sentences[i] >= 'A' && sentences[i] <= 'Z'))
        {
            m++;
        }
    }
    return m;
}
// no. of words
int words(string sentences)
{
    int m = 0;
    for (int i = 0; i < strlen(sentences); i++)
    {
        if(sentences[i] == ' ')
        {
         m++;
        }
    }
    return m+1;
}
// no. of sentences
int nsentence(string sentences)
{
    int m = 0;
    for (int i = 0; i < strlen(sentences); i++)
    {
        if(sentences[i] =='!' || sentences[i] == '.')
        {
            m++;
        }
    }
    return m;
}

File: readability.py
def main():
    #get user input
    sentences = input("provide: ")
    #calculate avg of the letters words and sentences
    avg_letters = (NumberOfLetters(sentences) / NumberOfWords(sentences)) * 100
    avg_sentences = (NumberOfSentences(sentences)/NumberOfWords(sentences)) * 100
    #calculate the grade using formula
    index = 0.0588 * avg_letters - 0.296 * avg_sentences - 15.8
    grade = round(index)
    #print the grade
    if grade <= 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade", grade)

#functions for finding number of letters
def NumberOfLetters(sentences):
    count = 0
    for i in range (len(sentences)):
        if sentences[i] >= 'a' and sentences[i] <= 'z' or sentences[i] >= 'A' and sentences[i] <= 'Z':
            count += 1
    return count

def NumberOfWords(sentences):
    count = 1
    for i in range(len(sentences)):
        if sentences[i] == ' ':
            count += 1
    return count

def NumberOfSentences(sentences):
    count = 0
    for i in range (len(sentences)):
        if sentences[i] == '.' or sentences[i] == '?' or sentences[i] == '!':
            count += 1
    return count

#calling main function
main()